### PR TITLE
Multiple assignments of workspaces to outputs (#555)

### LINKED ISSUE #555: multiple "workspace n output <foo>"
[Originally reported by c+i3@…]
(It is possible to have multiple workspace entries in the config, with different outputs. Only the last one is used.

-> Throw an error if there is already an output defined for a workspace
e.g.:

workspace 1 output HDMI1
workspace 1 output VGA1

workspace 1 will appear on VGA1 or on current if VGA1 is inactive. The line with HDMI1 is ignored

### PR DESCRIPTION
**Pull request by orestisf1993**
- [x] userguide
- [ ] I followed the discussion in #555 and concluded to use the `workspace X output A B C` syntax but it might be inconsistent with https://i3wm.org/docs/userguide.html#_output_s.
- [X] Fix infinite reccursion error
- [X] Fix mouse over crash
- [X] move_workspace_to_output crash

CC @ograff

Fixes #555.

### CODE REVIEW

**Comment by foonathan** [comment]
What is the status of this PR? I'd love to have this feature.

**Comment by orestisf1993** [comment]
Mainly needs to be reviewed. I didn't use the feature much on my machine so if you want to run this PR locally you might be able to spot possible issues since this change is a bit tricky.

**Comment by orestisf1993** [comment]
@Airblader I see that you put the `waiting` label. The only thing missing AFAIK is updating the userguide and writing some missing docstrings. I can do either if you want them before reviewing.

**Comment by Airblader** [comment]
The changes look fine to me at first glance, so I think we can go ahead.

**Comment by orestisf1993** [comment]
up

File: docs/userguide [hunk 1]
```diff
@@ -888,7 +888,7 @@ the second screen and so on).
 
 *Syntax*:
 -------------------------------------
-workspace <workspace> output <output>
+workspace <workspace> output <outputs…>
```
Comment (Airblader) [inline docs/userguide]: We write this as `[<output>+]` in other places in the userguide. I think we should use the same pattern here.

Comment (orestisf1993) [reply docs/userguide]: `[<XXX>+]` is only used for things that need a '+' character on the end right now, eg `[<Modifiers>+]` could be `Mod1+` or `Mod1+Shift+`. I think we should think of a new notation to not confuse users as outputs are not separated by pluses but by spaces.

Comment (orestisf1993) [reply docs/userguide]: But, tbf, no-one will really be confused since we'll have an example right after.

Comment (Airblader) [reply docs/userguide]: You're right, but nonetheless we have this case already in https://i3wm.org/docs/userguide.html#_manipulating_layout. Following that it should be `<output>…`. :-)

Comment (orestisf1993) [reply docs/userguide]: Maybe `<output1> [output2]…` since you have to specify at least one output.

Comment (Airblader) [reply docs/userguide]: GTM

**Pull request merged**